ros2_serial_example/CMakeLists.txt

```diff
@@ -99,22 +100,43 @@ target_link_libraries(bridge_gen
   fastcdr
 )
 
-add_executable(ros2_to_serial_bridge
+add_executable(ros2_to_serial_bridge_node
   src/ros2_to_serial_bridge.cpp
   src/ros2_to_serial_bridge_main.cpp
   src/uart_transporter.cpp
   src/udp_transporter.cpp
 )
-ament_target_dependencies(ros2_to_serial_bridge
+ament_target_dependencies(ros2_to_serial_bridge_node
   rclcpp
   ros2_serial_msgs
+  rclcpp_components
+)
+target_link_libraries(ros2_to_serial_bridge_node
+  bridge_gen
+  fastcdr
+  ring_buffer
+  transporter
+)
+
+add_library(ros2_to_serial_bridge SHARED
+  src/ros2_to_serial_bridge.cpp
+  src/uart_transporter.cpp
+  src/udp_transporter.cpp
 )
+target_compile_definitions(ros2_to_serial_bridge
+  PRIVATE "COMPOSITION_BUILDING_DLL")
```

Comment:
The visibility define should be custom to the package in question, rather than `COMPOSITION`

Reply:
Actually, you shouldn't need to do anything with visibility unless you are expecting to use this as a non-component.  The composition macros should take care of all the visibility needed to load as a component.